frontend/docsettings.lua

```diff
@@ -286,6 +286,7 @@ function DocSettings:open(doc_path)
         new.data = {}
     end
     new.data.doc_path = doc_path
+    new.data.partial_md5_checksum = hash_path_cache[doc_path]
```

Comment:
Feels (just reading these lines) like this should be better named `new.data.hash_path` ? Or `hash_path_cache` rename, if what it contains are not really paths but just hashes?

Reply:
`partial_md5_checksum` is an old name, exists in many books sdr and is used by KOSync.

Reply:
Yes, `doc_partial_md5_hash_cache` is what it is. Or just `hash_cache`.

Reply:
Oh right, that's `new.data`, so the docsetting stuff we know- ok then. (Was confusing it with the `new.hash_sidecar_file` we discussed some days ago.)

So, if it's a plain md5, its cache should have another name than `hash_path_cache`, no? `doc_hash_cache`?

Reply:
I've added this line to avoid recalculation of md5 in ReaderUI, if we have already done it with "hash" location.
But if we want to preserve access to the statistics (based on the doc_settings) even if the document has been edited (and md5 changed), this line should be removed, to not update the md5 value saved in doc_settings on the first opening.

Reply:
> I've added this line to avoid recalculation of md5 in ReaderUI, if we have already done it with "hash" location.

But, aren't you then overriding `stored.partial_md5_checksum` with `nil` if we are not using hashbased stuff and `hash_path_cache` stays empty ?

(Also, I don't think we really need to make any optimisation and too many tricks for/from the new hashbased stuff, it might be our newly adoptive child, but it's a crippled and ugly child, so best to hide it from the neighbours :)